Compentio.SourceMapper.Generator/Metadata/SourcesMetadata.cs

```diff
@@ -59,5 +61,66 @@ public void AddOrUpdate(IMapperMetadata mapperMetadata)
                 _mappers.Add(mapperMetadata);
             }
         }
+
+        /// <summary>
+        /// Collection of referenced assemblies <see cref="IAssemblySymbol" />
+        /// </summary>
+        private IReadOnlyCollection<IAssemblySymbol> _referencedAssemblies => _assemblySymbols?.Where(a => a.Identity?.HasPublicKey == false).ToList().AsReadOnly();
+
+        /// <summary>
+        /// Collection of referenced mappers <see cref="MapperMetadata" />
+        /// </summary>
+        private IReadOnlyCollection<IMapperMetadata> _referencedMappers
+        {
+            get
+            {
+                var nSpaceCollection = new List<INamespaceSymbol>();
+
+                foreach (var assembly in _referencedAssemblies)
+                {
+                    nSpaceCollection.AddRange(GetNamespaces(assembly.GlobalNamespace.GetNamespaceMembers()));
+                }
+
+                var typesCollection = nSpaceCollection?.SelectMany(n => n.GetTypeMembers());
+                var mappersCollection = typesCollection?.Where(t => t.GetAttributes().Any(attribute => attribute is not null && attribute.AttributeClass?.Name == nameof(MapperAttribute)));
+
+                return mappersCollection?.Select(t =>
+                {
+                    return new MapperMetadata(t, true);
+                }).ToList().AsReadOnly();
+            }
+        }
+
+        /// <summary>
+        /// Method returns recursively flatt collection of <see cref="INamespaceSymbol" />
+        /// </summary>
+        /// <param name="namespaceSymbols"></param>
+        /// <returns></returns>
+        private IEnumerable<INamespaceSymbol> GetNamespaces(IEnumerable<INamespaceSymbol> namespaceSymbols)
```

Comment:
Can we do it as in: https://github.com/alekshura/SourceMapper/blob/e640f2529e38020f585b899e4c5e6b8ee931d0d0/Compentio.SourceMapper.Generator/Metadata/TypeMetadata.cs#L72

Reply:
Updated.